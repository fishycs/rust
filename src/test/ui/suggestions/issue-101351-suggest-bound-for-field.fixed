// run-rustfix
// check-only

trait Foo {
    type Associated;
}

struct Generic<T: Foo> {
    field: T::Associated, //~ ERROR associated type `Associated` not found
}

fn main() {}
