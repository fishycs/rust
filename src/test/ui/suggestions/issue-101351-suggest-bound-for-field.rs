// run-rustfix
// check-only

trait Foo {
    type Associated;
}

struct Generic<T> {
    field: T::Associated, //~ ERROR associated type `Associated` not found
}

fn main() {}
