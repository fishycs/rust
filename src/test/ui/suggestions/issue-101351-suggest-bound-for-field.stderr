error[E0220]: associated type `Associated` not found for `T`
  --> $DIR/issue-101351-suggest-bound-for-field.rs:9:15
   |
LL |     field: T::Associated,
   |               ^^^^^^^^^^ there is a similarly named associated type `Associated` in the trait `Foo`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0220`.
